sdk/core/abort-controller/src/AbortSignal.ts

```diff
@@ -155,7 +155,10 @@ export function abortSignal(signal: AbortSignal): void {
 
   const listeners = listenersMap.get(signal)!;
   if (listeners) {
-    listeners.forEach((listener) => {
+    // Create a copy of listeners so mutations to the array
+    // (e.g. via removeListener calls) don't affect the listeners
+    // we invoke.
+    [...listeners].forEach((listener) => {
```

Comment:
Style nit: if you're using fancy `...` syntax, using `forEach` feels dated, expect `for...of` instead.

But if this is a low-level thing that we want to make sure is compatible in old moldy browsers, perhaps doing something like:

```suggestion
    listeners.slice().forEach((listener) => {
```

I don't have a strong preference, just mixing the two different eras of JS felt odd

Reply:
Interesting, I actually thought `[...listeners].forEach` looked nicer than something like `for (const listener of [...listeners])`, but I'll switch to `slice()`.